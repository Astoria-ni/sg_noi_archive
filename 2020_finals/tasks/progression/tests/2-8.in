25 1000
-9 -10 -9 10 3 -9 8 -8 -5 -10 -8 -8 7 -2 1 1 -8 -7 0 -3 -4 2 -4 3 4
3 1 25
1 14 23 -9 5
3 23 25
1 21 23 7 10
2 8 21 -9 -7
3 21 23
3 20 21
2 2 14 10 10
2 25 25 -8 -8
2 22 22 -9 -6
1 19 25 2 -5
2 9 16 6 3
3 22 25
1 9 19 9 4
1 8 19 4 -3
3 25 25
2 10 14 3 -2
1 25 25 1 10
1 21 22 5 6
2 17 18 -3 -10
2 10 15 -8 2
3 19 20
3 11 18
3 8 25
1 20 22 8 -8
3 16 25
2 7 25 4 6
2 16 18 2 -5
2 9 11 -4 9
2 3 20 8 -2
3 9 24
1 23 23 -1 -3
3 6 10
3 23 25
2 6 9 -1 2
3 4 4
2 10 10 7 -6
1 21 21 -4 10
1 13 13 -4 5
2 4 5 -8 -4
2 2 4 0 10
1 20 20 8 10
3 4 6
1 1 22 9 9
1 7 17 3 -6
1 13 24 10 5
2 24 24 -10 8
1 21 23 -7 -1
3 4 8
1 2 12 9 6
1 16 18 -6 2
1 23 25 -7 10
2 22 22 1 3
1 20 21 -2 3
2 19 20 -2 6
2 21 23 8 -9
3 20 23
2 16 18 4 -2
1 24 25 -6 3
2 23 23 -10 -2
2 11 19 6 -1
3 14 17
3 3 23
1 16 21 -3 -5
2 19 24 2 10
1 25 25 7 6
1 12 15 2 -1
2 24 25 3 -2
3 20 22
2 8 19 -8 1
2 12 14 -2 2
2 1 12 8 6
2 5 6 -6 -6
3 24 24
1 22 25 5 6
1 10 15 -4 2
2 1 19 9 10
3 11 24
2 8 22 -3 -9
2 20 23 0 6
1 5 8 -6 -4
1 14 24 10 9
1 2 16 -10 -4
2 15 19 4 7
3 12 18
2 14 16 2 8
3 24 24
2 23 23 4 1
3 6 6
1 22 25 6 -1
2 1 13 -6 9
1 6 22 10 -6
3 9 19
3 2 9
2 25 25 -1 0
1 11 17 8 8
2 13 23 9 8
2 21 22 -3 -2
1 10 11 -8 8
2 1 14 -2 5
2 22 23 -4 -1
2 24 25 -3 -2
3 3 12
3 1 3
3 22 22
3 14 16
2 23 25 -5 -9
1 11 17 -10 1
1 18 20 0 5
3 6 7
3 24 25
3 8 10
3 20 23
3 20 22
1 2 20 -8 5
1 21 24 -8 -2
3 25 25
1 22 24 6 -6
1 9 23 10 9
1 6 24 -7 -1
3 8 18
1 15 20 9 4
1 20 23 3 -4
2 20 21 -6 -4
3 21 25
2 1 1 8 1
2 6 19 1 -4
3 25 25
3 3 19
3 4 13
3 6 10
1 17 17 -1 0
3 16 19
1 13 24 1 6
3 24 25
3 15 17
1 8 21 3 9
1 1 17 9 7
3 1 10
1 2 4 5 -1
3 19 20
2 19 24 10 3
3 8 23
3 9 23
3 11 11
1 23 24 4 -5
2 7 7 -2 -1
2 12 14 -9 -1
3 6 15
3 9 13
2 4 11 1 10
2 13 22 -9 8
3 4 6
1 15 25 -9 0
1 17 25 -1 -2
2 9 9 -4 10
2 25 25 -10 6
2 25 25 2 -2
2 10 10 1 5
2 3 3 7 3
2 20 22 -5 8
3 19 19
3 17 18
1 22 23 8 5
1 8 18 -3 -1
2 3 17 -8 -9
1 21 25 -5 9
2 6 24 6 8
2 15 20 6 -3
1 6 12 9 -1
3 22 25
2 6 23 -1 10
1 25 25 3 -4
1 5 25 -1 7
2 3 5 -2 9
1 6 21 -3 -3
3 21 21
2 25 25 3 8
1 19 22 -8 -2
2 2 17 7 -10
1 23 23 -10 5
2 7 14 -6 3
2 24 24 -1 9
1 2 22 -4 9
3 7 24
1 1 6 6 -4
1 6 14 -1 3
1 8 8 9 0
2 7 19 -10 2
2 23 25 -10 0
2 10 19 -2 0
3 13 19
2 4 16 -6 4
2 15 20 -7 3
1 5 12 1 4
1 22 23 -2 -8
1 17 20 -2 -9
1 14 24 2 4
1 11 13 -8 4
2 4 4 10 0
2 5 12 -1 -10
2 19 20 -1 -6
3 22 23
2 5 13 -5 -10
2 23 25 -7 -4
1 2 11 -1 8
1 11 15 0 -4
1 5 9 6 -1
3 10 21
1 18 22 5 -8
3 14 20
1 5 23 -8 -9
1 23 24 7 -2
1 17 23 -9 -9
1 18 20 -6 -7
1 17 25 3 6
3 12 13
1 25 25 1 -8
3 1 16
1 8 13 8 7
1 20 24 -7 1
2 6 20 9 3
1 9 11 -6 -1
2 1 18 -6 -6
3 4 24
1 6 14 7 7
1 17 23 5 10
3 19 25
3 1 16
2 4 24 -7 8
2 9 9 -9 -9
2 6 25 10 5
3 17 19
2 15 16 10 -2
3 17 22
2 2 22 2 -7
3 1 8
1 15 20 3 -4
3 17 21
2 22 23 4 -5
1 9 25 5 8
3 16 19
3 8 23
3 22 22
2 6 10 0 -10
2 11 24 3 4
1 18 23 10 6
2 14 20 -2 -5
3 2 9
2 1 1 6 -8
1 20 23 -3 8
2 11 23 -1 -2
2 24 24 -10 -4
1 21 25 1 -8
2 8 25 7 -10
2 19 22 -6 -2
2 10 24 3 6
3 3 5
3 6 25
2 15 16 -10 -5
3 18 20
2 21 24 10 -9
3 23 23
1 16 22 0 -6
1 22 22 -5 8
1 13 14 -6 -4
1 8 23 0 5
3 12 14
1 23 24 -6 -10
2 9 18 -10 -6
1 12 12 9 6
1 8 11 4 3
1 11 25 3 0
1 18 22 -8 8
1 20 25 7 8
2 17 21 -5 -6
3 9 23
2 23 25 -1 3
1 4 21 -9 -8
2 15 21 5 -3
2 12 22 -10 1
1 6 23 6 1
2 13 15 2 -9
3 6 8
3 22 23
3 13 15
2 7 20 -10 -7
3 25 25
3 11 20
3 10 12
2 18 21 2 -10
1 18 19 -1 -10
2 3 11 8 0
1 6 10 -1 -5
2 7 8 -10 8
3 8 13
1 16 24 -4 -4
3 21 21
2 23 23 -4 6
3 23 24
2 21 21 6 4
3 12 16
1 23 23 2 0
2 22 22 2 3
1 19 24 -8 6
2 9 21 -7 7
2 6 11 -9 -9
3 21 22
1 15 23 2 -2
1 1 10 3 -4
1 3 3 -6 -6
1 2 2 -10 -10
2 5 14 -5 6
3 13 20
1 18 20 9 -1
3 20 22
1 23 23 6 -6
1 12 13 -10 -5
2 15 22 -7 7
3 4 16
2 17 20 10 5
1 5 15 -6 6
1 23 24 8 -1
1 11 25 0 -8
2 5 8 6 3
3 4 23
2 19 24 -9 9
1 24 24 -1 10
1 21 21 8 4
2 18 23 -6 10
1 6 7 -3 -7
1 14 20 3 1
1 8 11 -1 1
3 12 18
2 21 21 -10 9
3 11 12
1 19 19 -2 9
1 7 11 -2 -6
1 11 21 6 -3
3 12 23
3 25 25
2 4 16 -6 -10
3 8 21
3 14 21
2 1 22 2 9
2 10 12 -1 -3
1 12 23 -2 5
2 25 25 -3 -8
3 4 14
1 19 24 10 -6
2 5 23 0 6
3 23 25
1 17 17 -1 -7
1 21 23 -1 -6
1 19 25 9 -6
3 11 17
1 13 16 1 -3
3 13 15
3 14 18
2 3 4 -8 -7
1 1 22 -2 6
2 10 17 1 6
1 8 10 -10 -1
2 23 25 -9 6
3 14 21
3 7 21
2 2 12 -3 -1
1 13 19 -2 -1
2 21 22 8 6
2 4 15 -9 5
2 13 25 -8 4
2 6 18 3 10
3 9 11
1 1 4 8 3
1 18 25 1 -1
1 11 19 8 9
3 18 19
3 24 24
1 15 16 1 6
1 25 25 -3 5
3 3 4
2 14 20 -4 7
1 18 20 -5 -3
2 10 11 -3 7
2 6 21 -4 -10
2 19 24 -2 10
1 19 23 -1 6
1 17 21 -8 7
2 24 24 6 -7
1 3 15 2 5
3 1 13
1 10 19 -9 8
1 12 18 10 0
1 21 21 9 -7
1 5 23 8 10
3 21 21
3 9 25
2 22 25 7 4
2 23 23 -10 0
3 22 24
1 22 22 -4 7
3 13 24
2 4 19 4 -10
1 21 21 5 -1
1 22 23 -9 -4
3 9 17
1 11 17 3 -5
3 21 23
1 6 6 -10 2
2 17 23 10 9
3 3 6
3 5 13
3 21 24
3 16 20
2 16 20 -3 -3
3 19 21
3 1 13
2 15 16 -7 -2
3 8 17
3 6 21
1 20 22 -1 8
2 8 20 -1 0
2 18 25 -1 -1
1 3 25 7 5
1 18 18 4 6
1 6 17 4 -7
3 17 25
2 23 24 -4 1
2 2 25 8 -8
1 18 18 5 -8
3 11 20
2 10 12 -6 5
1 3 10 -10 2
3 7 15
1 16 21 4 3
3 12 14
3 25 25
3 23 25
2 23 25 2 -5
1 24 24 -3 9
2 17 19 -5 -7
1 22 23 -7 7
1 4 4 -2 2
3 3 21
1 5 12 10 -9
2 4 7 -4 -5
2 24 24 0 7
3 23 23
3 11 16
1 16 16 -4 4
2 23 24 -8 -1
2 2 16 8 5
2 18 21 -7 4
1 2 2 -2 -4
1 7 24 -10 -4
3 21 21
1 14 20 1 -9
2 17 17 6 9
1 3 3 -5 -2
2 3 21 1 5
2 25 25 7 -9
3 16 20
3 10 24
3 8 15
3 7 25
3 17 23
3 17 24
1 25 25 0 -10
2 2 14 -3 -7
3 20 20
2 23 23 10 1
3 16 20
1 23 25 8 -1
3 12 21
2 4 6 -4 -3
1 18 18 -2 -2
1 19 21 -4 -5
3 13 14
3 5 22
2 12 14 6 -9
3 9 16
3 10 24
2 2 2 9 1
3 23 25
3 13 24
3 16 25
3 19 19
2 16 25 -1 2
1 8 18 7 1
2 14 17 -7 0
1 8 24 -8 -10
2 19 22 -1 7
3 11 16
3 2 5
3 22 23
2 13 25 -4 9
2 25 25 -4 -8
1 12 23 7 7
2 2 20 -4 -3
3 11 15
3 10 21
1 21 25 -3 0
1 2 22 1 -3
1 13 19 -2 6
3 20 20
3 17 17
1 13 18 -9 1
3 5 10
3 16 22
2 6 11 6 -2
3 5 18
3 11 24
3 6 22
1 7 23 -10 -3
2 16 19 -7 7
2 10 20 0 -6
2 9 22 1 -10
2 19 21 5 7
3 20 23
1 18 18 4 4
2 24 25 10 8
1 11 18 -8 7
3 3 12
2 1 11 -4 4
1 23 24 -7 -6
1 5 15 4 2
2 8 15 -7 9
1 2 20 5 -7
1 5 20 -9 -6
3 7 15
2 8 8 1 -1
3 22 22
3 3 14
1 4 21 4 -9
2 16 24 -9 1
1 4 5 8 1
3 2 8
3 13 20
1 1 13 0 4
3 8 25
3 4 23
1 15 19 -5 -5
3 22 25
3 7 15
1 6 21 8 -6
3 17 19
1 25 25 8 -10
3 12 22
1 24 24 5 -5
1 16 21 -8 -8
2 25 25 3 -5
3 13 17
3 13 22
1 10 10 -1 -7
2 5 25 0 -7
2 2 12 0 -10
1 12 15 6 2
2 11 25 4 -2
3 18 24
3 15 19
3 5 12
2 6 7 9 7
1 12 20 -1 -3
1 19 22 -10 7
2 22 24 -8 -7
1 13 24 -4 -4
2 18 25 -3 5
2 12 15 2 7
3 20 23
1 6 14 10 8
2 20 21 5 -10
1 7 7 -8 -6
2 17 25 -6 6
1 8 16 -7 -3
2 2 22 1 10
3 13 21
1 9 10 -5 9
1 6 15 1 8
2 19 20 1 7
3 25 25
1 4 12 6 -7
2 1 8 1 -10
2 1 4 0 3
3 3 23
1 19 20 -1 1
3 15 15
2 1 17 4 10
2 20 25 10 3
1 10 10 -6 -4
1 1 8 -10 0
2 2 14 5 7
1 9 23 10 -9
2 16 19 9 3
3 21 22
2 24 25 -8 -7
3 7 25
1 15 19 -6 -7
3 4 17
3 13 21
2 6 18 -7 5
2 10 17 -6 2
2 3 18 3 2
1 13 14 2 -10
3 20 24
2 12 12 -6 6
2 2 7 -2 -8
1 11 24 -10 -7
1 19 20 -4 -4
1 21 21 0 9
3 6 14
1 19 24 -8 6
3 14 17
1 15 25 -3 9
3 7 24
3 9 16
2 18 23 -6 -2
1 15 22 -9 3
1 3 3 8 -6
1 4 4 -5 5
1 22 25 -5 -10
3 9 10
3 1 2
1 16 18 0 6
3 19 21
2 2 9 -4 -2
2 12 23 2 3
1 6 20 8 7
3 19 23
1 11 23 9 -4
1 7 8 -9 4
1 13 15 -4 0
3 18 24
2 8 13 -9 3
3 4 4
2 22 23 5 -9
3 23 23
2 16 24 -2 -2
1 9 20 -1 1
3 25 25
2 12 21 -5 3
1 13 18 10 4
1 24 24 -7 3
2 21 23 2 5
2 23 23 -9 9
1 11 25 -3 -6
1 12 14 9 -6
1 14 15 -9 -4
1 18 18 -6 6
3 11 18
1 4 25 0 9
1 21 24 -5 9
1 11 14 3 -2
2 2 22 3 -1
2 1 22 -5 5
1 16 17 4 0
1 11 14 5 -9
3 16 20
2 13 20 -4 -1
1 22 22 8 5
3 15 23
2 6 22 3 7
1 17 24 -1 3
2 22 25 4 -4
2 7 20 -3 -5
1 10 21 8 9
1 23 25 3 -9
1 5 25 2 5
2 10 25 -1 2
1 18 21 8 -8
1 17 19 5 -10
1 1 25 -1 -3
3 12 13
3 10 15
1 12 22 -1 -4
1 3 9 8 8
3 9 12
2 25 25 10 -3
2 4 25 -3 -4
2 1 9 -6 -10
3 1 22
1 24 24 5 -7
2 3 14 -3 -7
2 8 18 -4 -2
1 19 23 4 -10
1 14 18 10 -4
3 16 17
1 10 10 1 3
3 1 15
3 21 24
1 14 14 2 6
1 15 18 -9 -7
1 23 24 -7 -4
2 10 25 5 3
1 4 4 -3 5
2 20 20 -10 6
1 17 17 -2 7
1 4 15 -1 -4
2 2 17 -1 -4
1 15 24 -9 2
3 21 23
1 5 5 6 -2
3 25 25
3 24 25
3 13 21
3 12 25
1 16 22 7 -3
1 17 20 -1 0
2 18 25 -3 9
1 6 25 9 -1
3 1 23
1 17 23 2 -7
2 4 17 -8 6
1 11 21 6 1
1 1 16 8 1
2 14 17 4 4
2 20 23 10 4
2 1 9 -2 0
1 22 24 -7 2
1 25 25 4 -2
2 12 13 -7 -5
2 6 16 6 -5
1 19 25 8 5
1 15 20 2 -4
3 5 16
3 9 9
2 14 16 1 -5
3 5 9
3 8 10
3 22 24
3 13 14
3 2 20
1 6 6 8 9
3 20 21
3 21 21
2 16 22 4 3
1 3 19 2 1
2 16 24 1 1
3 16 21
1 5 11 0 -5
2 17 25 -8 -4
2 25 25 -6 -4
3 18 23
1 24 25 0 4
3 4 25
2 21 21 6 -8
2 3 4 6 2
1 19 22 -4 4
1 12 23 4 -10
2 20 24 -4 2
1 3 8 -8 2
3 1 16
3 21 21
1 11 22 -7 7
1 11 15 -9 2
2 15 25 -10 2
1 12 18 2 -7
1 7 11 0 -3
1 9 13 4 10
2 5 16 -5 10
3 2 14
3 10 12
1 14 25 -3 -7
2 12 14 1 -10
1 1 2 -8 0
3 8 19
3 13 25
3 10 25
3 20 25
2 3 12 -4 7
1 4 17 1 -4
2 5 22 5 3
3 23 23
3 23 25
1 12 21 10 -3
1 13 23 -7 -5
1 18 25 3 0
2 5 7 -4 -8
2 12 12 -10 8
2 14 14 8 0
1 2 2 10 -4
1 25 25 2 6
1 22 24 -2 0
3 18 18
1 21 25 10 8
1 23 24 -8 -10
3 4 25
3 14 22
2 1 2 5 10
1 1 15 5 8
2 11 23 2 -9
3 1 22
3 11 11
1 24 25 -10 -2
2 11 25 -10 10
2 4 19 -3 5
1 20 25 6 -8
3 25 25
3 24 24
1 23 25 5 -4
1 25 25 -6 -1
2 10 19 -5 -4
1 10 23 -5 -9
1 15 17 7 -1
1 19 20 -3 8
3 16 18
1 14 21 5 -6
1 25 25 3 8
2 24 24 6 -2
3 18 22
1 20 23 5 2
3 19 20
1 10 12 -10 -6
2 20 25 -4 -5
2 22 24 -7 -8
1 24 24 2 -10
3 24 24
1 13 21 2 7
1 14 17 -4 5
3 24 25
2 18 20 -7 1
2 25 25 8 8
1 3 11 -1 8
3 5 7
1 7 8 -2 7
3 24 24
1 21 23 -6 -4
3 25 25
1 1 4 6 -2
2 16 22 8 2
1 12 21 -2 0
3 9 25
1 10 15 -7 -1
3 6 14
3 21 22
2 13 13 3 6
2 5 23 5 -5
2 1 16 -7 -6
3 25 25
1 25 25 -1 -8
2 22 24 -6 5
1 19 21 10 6
2 13 21 -9 -6
2 19 20 -9 -9
2 9 11 -1 -4
2 8 15 -6 10
2 2 6 -9 -7
1 22 22 -4 0
3 5 22
1 23 23 -8 1
3 11 12
2 12 18 6 6
1 23 25 8 8
3 21 23
3 19 25
2 16 22 9 4
3 16 20
3 22 25
3 11 25
3 10 20
1 13 14 -7 6
3 11 11
1 17 24 -4 9
3 13 25
1 20 24 -10 4
3 23 25
2 15 24 6 -5
1 21 25 4 -8
3 14 25
2 8 20 -2 -8
3 14 14
3 3 5
2 23 25 8 0
1 5 10 4 -5
3 7 25
3 7 24
2 23 23 -1 -1
1 22 24 -2 -3
1 15 16 -5 -10
3 25 25
1 6 24 7 5
2 17 18 -9 -3
3 11 12
1 5 5 -5 2
1 18 18 -3 4
2 2 2 -6 -10
1 25 25 -7 2
1 5 17 3 0
1 4 22 6 6
1 1 15 -5 10
2 21 21 8 8
3 22 25
1 4 4 -6 -8
3 16 23
2 14 22 -3 -10
2 7 16 -9 -8
2 3 8 -7 -2
3 20 22
2 23 25 1 4
1 7 8 10 4
2 14 19 5 4
1 17 22 7 -10
1 25 25 -4 -6
1 12 20 -5 -8
2 17 18 -3 -6
3 12 12
1 23 23 2 -2
1 18 19 -9 7
2 1 25 7 -6
1 12 22 9 -5
1 8 13 -6 -10
1 12 14 -10 6
2 25 25 -2 5
2 1 1 -1 2
1 9 9 10 -8
1 18 21 10 -10
1 21 22 -8 -2
3 3 5
2 16 25 3 -10
2 10 16 -1 2
2 9 13 3 -5
3 1 16
2 22 25 -6 5
3 4 23
1 1 1 -2 9
1 7 11 10 2
3 5 6
2 14 17 -10 -7
1 11 24 1 7
1 24 24 -8 -2
3 6 9
1 6 18 3 6
2 16 16 -1 -6
2 10 13 0 6
2 14 14 -2 1
2 11 25 -4 -1
1 12 16 -4 -8
2 14 23 10 2
2 17 18 8 -2
1 13 14 -2 8
2 15 20 -9 -8
2 12 17 4 -7
3 21 23
3 19 20
1 16 22 -1 -9
2 19 24 -9 2
3 6 19
1 18 24 -2 1
2 13 21 -3 -5
3 21 21
2 19 22 0 -10
3 22 24
1 4 22 8 0
1 19 25 -7 3
3 16 21
3 10 17
2 14 14 2 2
1 1 1 5 10
2 10 24 -1 8
1 20 20 1 5
1 17 21 4 -2
3 1 20
1 3 11 6 6
3 3 25
3 6 23
3 7 23
2 9 16 -10 2
1 5 20 -6 -9
2 6 20 4 -4
2 8 23 -7 -1
3 15 15
3 16 18
3 14 17
2 3 24 -1 3
2 5 16 1 -6
2 13 13 4 -4
2 22 25 -3 0
1 6 25 0 -8
2 3 15 0 -2
1 16 25 1 9
2 5 7 1 -9
1 8 10 1 -5
2 24 24 9 -5
2 13 14 -4 -7
2 21 23 9 -6
2 22 23 2 7
3 13 18
3 21 22
2 23 24 -9 6
2 3 3 1 -2
2 15 22 7 -9
1 22 24 -3 -10
1 8 8 7 1
3 20 20
3 17 21
1 21 21 -6 -6
2 19 23 -3 -10
1 20 24 -6 1
2 22 24 -9 9
3 14 21
3 1 25
