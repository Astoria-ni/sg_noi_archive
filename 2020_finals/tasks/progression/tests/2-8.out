2
2
2
2
2
1
2
5
6
2
12
5
2
1
2
2
3
4
9
3
1
9
4
1
1
5
4
10
3
1
2
2
2
3
3
2
1
4
2
1
14
8
5
2
2
2
4
2
5
5
1
4
3
3
1
2
3
1
8
7
2
2
2
2
5
15
2
10
3
6
7
4
4
6
1
4
3
16
3
1
2
5
2
2
3
1
10
3
2
1
2
4
2
6
2
7
4
2
2
2
1
9
3
6
2
6
3
3
4
7
3
2
1
2
8
1
4
2
3
9
2
2
4
3
4
2
4
4
4
7
7
3
2
1
3
5
1
3
1
5
12
8
15
5
5
1
5
7
2
8
3
3
2
3
3
1
3
2
2
5
11
1
1
6
3
6
6
6
2
3
8
1
6
2
5
9
8
3
5
3
4
2
4
7
5
6
4
9
1
4
1
2
6
6
4
4
3
3
4
2
2
2
3
3
3
1
1
1
2
3
6
2
6
2
13
2
5
3
3
1
2
3
4
9
9
1
4
3
2
2
8
2
1
6
6
8
3
1
10
3
4
7
7
6
1
3
1
4
3
12
1
1
1
2
2
2
1
2
3
1
1
6
2
2
1
8
2
2
2
5
2
7
5
1
6
2
6
1
3
10
10
1
2
2
3
3
1
3
6
5
2
2
3
2
4
1
2
3
5
7
5
5
5
1
3
4
2
2
1
5
4
4
